examples/graph/app.js

```diff
@@ -0,0 +1,393 @@
+      return 'fries';
+    case 2:
+      return 'soda';
+    case 3:
+    case 4:
+      return 'pie';
+    default:
+      return null;
+    }
+  }
+
+  //
+  // rendering
+  //
+  _renderInteractionLayer(viewport, hovered, clicked) {
+    // set flags used below to determine if SVG highlight elements should be rendered.
+    // if truthy, each flag is replaced with the corresponding element to render.
+    const elements = {
+      hovered: hovered && hovered.object,
+      clicked: clicked && clicked.object
+    };
+    const relatedElements = {
+      hovered: hovered && hovered.relatedObjects,
+      clicked: clicked && clicked.relatedObjects
+    };
+    const elementInfo = {
+      hovered: hovered && hovered.object,
+      clicked: clicked && clicked.object
+    };
+
+    // process related elements first, since they compare themselves to the focused elements
+    Object.keys(relatedElements).forEach(k => {
+      const els = relatedElements[k];
+      if (!els || !els.length) {
+        relatedElements[k] = null;
+      } else {
+        relatedElements[k] = [];
+        els.forEach(el => {
+          if (el.source) {
+            // highlight linked nodes, as well as the links
+            const relatedNode = el.source === elements[k] ? el.target : el.source;
+            relatedElements[k].push(
+              this._renderInteractionElement(relatedNode, `related ${k}`, viewport)
+            );
+          }
+          relatedElements[k].push(
+            this._renderInteractionElement(el, `related ${k}`, viewport)
+          );
+        });
+      }
+    });
+
+    // process the focused (hovered / clicked) elements
+    Object.keys(elements).forEach(k => {
+      const el = elements[k];
+      elements[k] = el ? this._renderInteractionElement(el, k, viewport) : null;
+    });
+
+    // render additional info about the focused elements (only nodes, not links)
+    Object.keys(elementInfo).forEach(k => {
+      const el = elementInfo[k];
+      if (el && el.name) {
+        elementInfo[k] = (<text
+          x={el.x + viewport.width / 2}
+          y={el.y + viewport.height / 2}
+          dx={this._getNodeSize(el) + 10}
+          dy={-10}
+        >{el.name}</text>);
+      } else {
+        elementInfo[k] = null;
+      }
+    });
+
+    return (
+      <svg width={viewport.width} height={viewport.height} className="interaction-overlay">
+        {relatedElements.hovered}
+        {elements.hovered}
+        {relatedElements.clicked}
+        {elements.clicked}
+        {elementInfo.hovered}
+        {elementInfo.clicked}
+      </svg>
+    );
+  }
+
+  _renderInteractionElement(el, className, viewport) {
+    // Note: node.x/y, calculated by d3 layout,
+    // is measured from the center of the layout (of the viewport).
+    // Therefore, we offset the node coordinates from the viewport center.
+
+    let element;
+    if (el.source) {
+      // link
+      element = (<line
+        x1={el.source.x + viewport.width / 2}
```

Comment:
instead of adding the offset to x and y for each element, we can keep move it out by adding a 
<g transform={\`translate(${viewport.width / 2} ${viewport.height / 2})\`}>{...}\</g>
that wraps the content in the <svg> tag in line 295.